### PR TITLE
fix: button state colors for safari browsers

### PR DESCRIPTION
**Pull request by rustem-nasyrov**
## Description
- Fix button colors on safari <16 versions.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Comment by rustem-nasyrov** [comment]
image
Oopsie. :D

**Review by aluarius — CHANGES_REQUESTED** [review]
Now plain states are broken in new (16.2) Safari :c

I guess we need to specify version at `getOpacity`.

File: packages/ui/src/components/va-button/hooks/useButtonTextColor.ts [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
+* @returns {number} Inverted opacity value
+* */
+const getOpacity = (opacity: number): number => {
+  if (isServer()) { return opacity }
+
+  if (/^((?!chrome|android).)*safari/i.test(window?.navigator?.userAgent)) {
+    const isLatestSafari = /(version.)15|16/i.test(window?.navigator?.userAgent)
+
+    if (!isLatestSafari) {
+      return opacity < 1 ? 1 - opacity : opacity
```
Comment (m0ksem) [inline packages/ui/src/components/va-button/hooks/useButtonTextColor.ts:37]: wtf?

Looks like it is impossible to have opacity 0 here. Is that ok?

Should be more documented, cuz for now it is magic.

**Review by m0ksem — APPROVED** [review]

File: packages/ui/src/components/va-button/hooks/useButtonTextColor.ts (hunk 1 above)
Comment (rustem-nasyrov) [reply packages/ui/src/components/va-button/hooks/useButtonTextColor.ts:37]: Oopsie, forgot about it. Will fix it. :)

**Review by aluarius — APPROVED** [review]
Gj! Now everything works fine everywhere!

**Pull request merged**